youtube_dl/utils.py

```diff
@@ -2496,41 +2496,143 @@ def srt_subtitles_timecode(seconds):
+                self._out += '\n'
+            else:
+                unclosed_elements = []
+                style = {}
+                element_style_id = attrib.get('style')
+                if default_style:
+                    style.update(default_style)
+                if element_style_id:
+                    style.update(styles.get(element_style_id, {}))
+                for prop in SUPPORTED_STYLING:
+                    prop_val = attrib.get(_x('tts:' + prop))
+                    if prop_val:
+                        style[prop] = prop_val
+                if style:
+                    font = ''
+                    for k, v in style.items():
+                        if self._applied_styles and self._applied_styles[-1].get(k) == v:
+                            continue
+                        if k == 'color':
+                            font += ' color="%s"' % v
+                        elif k == 'fontSize':
+                            font += ' size="%s"' % v
+                        elif k == 'fontFamily':
+                            font += ' face="%s"' % v
+                        elif k == 'fontWeight' and v == 'bold':
+                            self._out += '<b>'
+                            unclosed_elements.append('b')
+                        elif k == 'fontStyle' and v == 'italic':
+                            self._out += '<i>'
+                            unclosed_elements.append('i')
+                        elif k == 'textDecoration' and v == 'underline':
+                            self._out += '<u>'
+                            unclosed_elements.append('u')
+                    if font:
+                        self._out += '<font' + font + '>'
+                        unclosed_elements.append('font')
+                    applied_style = {}
+                    if self._applied_styles:
+                        applied_style.update(self._applied_styles[-1])
+                    applied_style.update(style)
+                    self._applied_styles.append(applied_style)
+                self._unclosed_elements.append(unclosed_elements)
 
         def end(self, tag):
-            pass
+            if tag not in (_x('ttml:br'), 'br'):
+                unclosed_elements = self._unclosed_elements.pop()
+                for element in reversed(unclosed_elements):
+                    self._out += '</%s>' % element
+                if unclosed_elements and self._applied_styles:
+                    self._applied_styles.pop()
 
         def data(self, data):
-            self.out += data
+            self._out += data
 
         def close(self):
-            return self.out.strip()
+            return self._out.strip()
 
     def parse_node(node):
         target = TTMLPElementParser()
         parser = xml.etree.ElementTree.XMLParser(target=target)
         parser.feed(xml.etree.ElementTree.tostring(node))
         return parser.close()
 
+    for k, v in LEGACY_NAMESPACES.items():
```

Comment:
i'm not sure if there is a btter way to fix the test, but this will make the tests pass:
```diff
diff --git a/test/test_utils.py b/test/test_utils.py
index 3895daf2f..038cb9b5a 100644
--- a/test/test_utils.py
+++ b/test/test_utils.py
@@ -14,6 +14,7 @@ sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
 import io
 import json
 import xml.etree.ElementTree
+import re
 
 from youtube_dl.utils import (
     age_restricted,
@@ -1089,7 +1090,24 @@ part 3</font></u>
  </font>style</font></u></i>
 
 '''
-        self.assertEqual(dfxp2srt(dfxp_data_with_style), srt_data)
+
+        def xml_to_dict(elem):
+            return {
+                'tag': elem.tag,
+                'text': elem.text,
+                'tail': elem.tail,
+                'attrib': elem.attrib,
+                'childrens': [xml_to_dict(child) for child in elem],
+            }
+
+        time_re = r'\d{2}:\d{2}:\d{2},\d{3}'
+        str_part_re = re.compile(r'(?s)(\d+)\n(%s) --> (%s)\n(<[a-z]+.+?[a-z]+>)\n\n' % (time_re, time_re))
+
+        for (num1, start1, end1, text1), (num2, start2, end2, text2) in zip(str_part_re.findall(dfxp2srt(dfxp_data_with_style)), str_part_re.findall(srt_data)):
+            self.assertEqual(num1, num2)
+            self.assertEqual(start1, start2)
+            self.assertEqual(end1, end2)
+            self.assertEqual(xml_to_dict(compat_etree_fromstring(text1)), xml_to_dict(compat_etree_fromstring(text2)))
 
     def test_cli_option(self):
         self.assertEqual(cli_option({'proxy': '127.0.0.1:3128'}, '--proxy', 'proxy'), ['--proxy', '127.0.0.1:3128'])
```